packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -189,6 +190,11 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
         return this._containerRuntime.disableIsolatedChannels;
     }
 
+    /** Tells whether GC data will be written at the root of the summary tree. If so, data store should not write it. */
+    protected get writeGCDataAtRoot(): boolean {
```

Comment:
same as above, do  this need to be exposed so generally. i don't really like the disableIsolatedChannels above either. should we be passing some object will all these feature facts on it downward?

Reply:
it doesn't seem like any child classes even needs this, so why is it a property at all?

Reply:
I agree with you. We should not pass container runtime to data store context at all. We should only pass the things that it needs as an object or a set of params. That is a bigger change though and will also help us in writing unit tests.
For now, I can pass the `writeGCDataAtRoot` property as a param. Since they are in same layer, there won't be compatibility concerns.

Reply:
Updated to pass this as a param to data store context.